### PR TITLE
Refactor the representation of intrinsic functions in the AST

### PR DESCRIPTION
**Pull request by emilyaherbert**

### CODE REVIEW

File: sway-core/src/control_flow_analysis/dead_code_analysis.rs [hunk 1]
```diff
@@ -1049,6 +1028,41 @@ fn connect_expression(
...
+        )?,
+        TypedIntrinsicFunctionKind::GetPropertyOfType { .. } => {
+            let node = graph.add_node("Type Property".into());
+            for leaf in leaves {
+                graph.add_edge(*leaf, node, "".into());
+            }
+            vec![node]
+        }
+        TypedIntrinsicFunctionKind::GetStorageKey => {
+            let node = graph.add_node("Generate B256 Seed".into());
```
Comment (mohammadfawaz) [inline sway-core/src/control_flow_analysis/dead_code_analysis.rs:1056]:
```suggestion
            let node = graph.add_node("Get storage key".into());
```
I forgot to change this when I replaced `__generate_uid` with `__get_storage_key`.

**Review by mohammadfawaz — APPROVED** [review]
Much cleaner! Thank you :)

**Review by mohammadfawaz — APPROVED** [review]
Awesome, thanks!

**Comment by AlicanC** [comment]
> > I was actually working on implementing the intrinsics I've mocked in this PR: https://github.com/FuelLabs/sway/pull/1627/files#diff-ad206f980045decc1462a5334f05035d01317ea4a0af85650d33556a7e8b1fb3R19-R48
> > Will rebase and take a look.
> 
> You're adding a library intrinsic though right? Not a compiler intrinsic. So you shouldn't really be affected.

I'm trying to turn those into real intrinsics.

**Pull request merged**